core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1019,7 +1019,11 @@ class SnappySession(@transient private val sc: SparkContext,
               userSpecifiedSchema = userSpecifiedSchema,
               partitionColumns = partitionColumns,
               options = params).write(mode, data)
-            (r, Some(r.schema))
+            if (None != userSpecifiedSchema) {
+              (r, Some(userSpecifiedSchema.get))
+            } else {
+              (r, Some(r.schema))
+            }
```

Comment:
@SachinJanani Can you explain why is this required (and even seem incorrect for some cases)? The DataSource.resolveRelation will lead to dataSource.createRelation with the userSpecifiedSchema so the schema of the underlying relation should be as provided. As far as I can see the schema of query is not used in this method anywhere.

Even otherwise if the schema of the returned relation is incorrect then that will lead to other problems and just changing the schema entry in Hive catalog will not be a proper fix.

Btw in other note, for something like above we should use scala-like way rather than java-like if-else etc: ```userSpecifiedSchema.getOrElse(Some(r.schema))```